telegram/utils/request.py

```diff
@@ -171,9 +171,15 @@ def _parse(json_data):
             decoded_s = json_data.decode('utf-8')
             data = json.loads(decoded_s)
         except UnicodeDecodeError:
-            logging.getLogger(__name__).debug(
-                'Logging raw invalid UTF-8 response:\n%r', json_data)
-            raise TelegramError('Server response could not be decoded using UTF-8')
+            try:
+                decoded_s = json_data.decode('utf-8', 'ignore')
+                data = json.loads(decoded_s)
+            except UnicodeDecodeError:
+                logging.getLogger(__name__).debug(
+                    'Logging raw invalid UTF-8 response:\n%r', json_data)
+                raise TelegramError('Server response could not be decoded using UTF-8')
+            except ValueError:
+                raise TelegramError('Invalid server response')
```

Comment:
1. no need to decode twice. just call:
```python
decoded_s = json_data.decode('utf-8', 'replace')
```
2. please use `replace` instead of `ignore`.
3. the `except UnicodeDecodeError` is not needed any longer

Reply:
Got it.😋